### PR TITLE
Restructure halo modules folder to centralise Covenant content

### PR DESCRIPTION
**Pull request by caelaislinn**
Maintenance PR to move around a bunch of files so they are sorted and easier to find + maintain. Due to the amount of content touched, there is a chance of bugs being introduced. The most likely content a bug could appear is in clothing. 

Some other unrelated changes snuck in, including
- Miscellaneous cleanups and fixes (see commit log)
- Ruutian and Tvoan rangers share the same equipment
- Job arrival announcements should be working now
- Flood Gravemind no longer issues orders over the public radio channel
- Rebalances faction NPC soldiers. Covenant should be fairly tougher now

:cl: CaelAislinn
experiment: Restructures the Covenant code. There should be no changes to gameplay from this but it may introduce bugs in Covenant equipment. Report bugs on github
rscadd: Tvoan and Ruutian rangers now share the same equipment. All 3 pieces are necessary to be spaceworthy undersuit, armour, helmet but gloves and boots are optional
rscadd: Covenant NPC soldiers now have much more diversity and should overall be more challenging adds Elite minors and majors, shield jackals
/:cl:

### CODE REVIEW

**Comment by caelaislinn** [comment]
After #2381 is merged I'll be able to fix the map compile failures. EDIT: done

File: code/modules/halo/covenant/simple_mobs/_covenant.dm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+/mob/living/simple_animal/hostile/covenant
+	faction = "Covenant"
```
Comment (X0-11) [inline code/modules/halo/covenant/simple_mobs/_covenant.dm:3]: Simple mobs intended for combat should have resistance set to simulate their armour. Defender mobs use 10, for reference.

File: code/modules/halo/covenant/species/kigyar/clothing.dm [hunk 2]
```diff
@@ -1,37 +1,11 @@
...
+#define KIGYAR_CLOTHING_PATH 'code/modules/halo/covenant/species/kigyar/jackalclothing.dmi'
 
 /obj/item/clothing/head/helmet/kigyar
 	name = "Kig-Yar Scout Helmet"
 	desc = "A Kig-Yar scout helmet with inbuilt night vision."
-	icon = 'code/modules/halo/icons/species/jackalclothing.dmi'
+	icon = KIGYAR_CLOTHING_PATH
 	icon_state = "scouthelm"
-	sprite_sheets = list("Kig-Yar" = 'code/modules/halo/icons/species/jackalclothing.dmi',"Tvaoan Kig-Yar" = 'code/modules/halo/icons/species/skirm_clothing.dmi')
+	sprite_sheets = list("Kig-Yar" = KIGYAR_CLOTHING_PATH)
```
Comment (X0-11) [inline code/modules/halo/covenant/species/kigyar/clothing.dm:8]: No longer supports Tvoans?

Comment (caelaislinn) [reply code/modules/halo/covenant/species/kigyar/clothing.dm:8]: Turns out the scout helm doesn't actually have a Tvoan on-mob sprite, and isn't in any Tvoan outfits. 

But then it doesn't have an on-ground state either so I could merge them seeing as they are similar... only problem is they look somewhat different

File: code/modules/halo/covenant/simple_mobs/_covenant.dm (hunk 1 above)
Comment (caelaislinn) [reply code/modules/halo/covenant/simple_mobs/_covenant.dm:3]: Yeah, by default I'm setting it to 0 resistance for those without armour. I feel like 10 is a bit high though for weak enemies though, so I'm testing 5 resistance for them and 0 resist for unarmoured.

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**